File: rmt-services/RMT.Identity/src/lib/user/updateUser.dto.ts
import { IsBoolean, IsOptional, IsString } from 'class-validator';

export class UpdateUserDto {
  @IsOptional()
  @IsString()
  roles: string;

  role_ids: string;

  // @IsOptional()
  // @IsString()
  // emp_id: string;

  @IsOptional()
  @IsString()
  emp_code: string;

  @IsOptional()
  @IsBoolean()
  status: boolean;
}

File: rmt-services/RMT.Gateway/Gateway.API/Middlewares/MiddlewareHelpers/RolesAndPermissionsMiddlewareHelper/DTOs/UpdateUserDto.cs
namespace Gateway.API.Middlewares.MiddlewareHelpers.RolesAndPermissionsMiddlewareHelper.DTOs
{
    public class UpdateUserDto
    {
        public string[]? roles { get; set; }
        public string? role_ids { get; set; }
        public string? emp_code { get; set; }
        public bool? status { get; set; }
    }
}
